client/src/app/routes/nova/OutputPage.tsx

```diff
@@ -47,6 +50,20 @@ const OutputPage: React.FC<RouteComponentProps<OutputPageProps>> = ({
 
     const { blockId, transactionId, outputIndex, isSpent, transactionIdSpent } = outputMetadataResponse ?? {};
 
+    // @ts-expect-error TODO: Remove this ignore once included/spent are honoured in the type https://github.com/iotaledger/iota-sdk/issues/1884
+    const createdSlotIndex = (outputMetadataResponse?.included?.slot as number) ?? null;
+    // @ts-expect-error TODO: Remove this ignore once included/spent are honoured in the type https://github.com/iotaledger/iota-sdk/issues/1884
+    const spentSlotIndex = (outputMetadataResponse?.spent?.slot as number) ?? null;
+
+    let outputManaDetails: OutputManaDetails | null = null;
+    if (output !== null && createdSlotIndex !== null && protocolInfo !== null) {
```

Comment:
we could simplify these checks with just `if (output && createdSlotIndex && protocolInfo)`
and also inside the if clause we could on only figure out the `slotIndex` and call `buildManaDetailsForOutputs` only once
along the line of:
```
if (output && createdSlotIndex && protocolInfo) {
    const slotIndex = isSpent && spentSlotIndex ? 
    spentSlotIndex :
    latestConfirmedSlot > 0 ? latestConfirmedSlot : null;
    outputManaDetails = slotIndex ? buildManaDetailsForOutput(output, createdSlotIndex, slotIndex, protocolInfo.parameters) : null;
}
```

Reply:
Updated this part a bit. But `slotIndex` cannot be allowed to be `null` as in the snippet above.

Reply:
In proposed snippet there is a check for slot index not being null `slotIndex ?` and only the do we call `buildManaDetailsForOutput` :thinking:

Reply:
My bad, overlooked the last ternary.